### PR TITLE
Print output type in PlanNode::toString

### PR DESCRIPTION
**Pull request by majetideepak**
Adds the output type of the plan node when `PlanNode::toString` is invoked with `detailed = true`.
Changes the prefix format of the output to `--`.

Example output
```
  -- Project[expressions: (out3:BIGINT, plus(cast ROW["out1"] as BIGINT,10))] -> out3:BIGINT
    -- Filter[expression: lt(mod(cast ROW["out1"] as BIGINT,10),8)] -> out1:SMALLINT, out2:BIGINT
      -- Project[expressions: (out1:SMALLINT, ROW["c0"]), (out2:BIGINT, plus(mod(cast ROW["c0"] as BIGINT,100),mod(cast ROW["c1"] as BIGINT,50)))] -> out1:SMALLINT, out2:BIGINT
        -- Filter[expression: lt(mod(cast ROW["c0"] as BIGINT,10),9)] -> c0:SMALLINT, c1:INTEGER, c2:BIGINT
          -- Values[5 rows in 1 vectors] -> c0:SMALLINT, c1:INTEGER, c2:BIGINT
```

### CODE REVIEW

File: velox/core/PlanNode.cpp [hunk 1]
```diff
@@ -433,12 +433,15 @@ void PlanNode::toString(
...
 
-  stream << indentation << "-> " << name();
+  stream << indentation << "-- " << name();
 
   if (detailed) {
     stream << "[";
     addDetails(stream);
     stream << "]";
+    stream << " -> ";
+    const auto& outputType = this->outputType();
```
Comment (mbasmanova) [inline velox/core/PlanNode.cpp:443]: nit: any particular reason for this temp variable? Why not simply `outputType()->printChildren(stream)`?

File: velox/type/Type.h [hunk 2]
```diff
@@ -698,6 +698,9 @@ class RowType : public TypeBase<TypeKind::ROW> {
 
   std::string toString() const override;
 
+  // Print children name and type as comma separated pairs.
```
Comment (mbasmanova) [inline velox/type/Type.h:701]: typo: "children name and type" -> "child names and types"

Use /// for method comments.

**Review by mbasmanova — APPROVED** [review]
@majetideepak Looks great. Thank you, Deepak, for improving the output of printPlanWithStats() and PlanNode::toString().

File: velox/type/Type.h (hunk 2 above)
Comment (majetideepak) [reply velox/type/Type.h]: Shouldn't it be `children names and types`?

**Comment by mbasmanova** [comment]
@majetideepak Deepak, there are CI failures. Would you take a look?

File: velox/type/Type.cpp [hunk 3]
```diff
@@ -330,13 +330,11 @@ bool RowType::operator==(const Type& other) const {
...
 }
 
-std::string RowType::toString() const {
-  std::stringstream ss;
-  ss << (TypeTraits<TypeKind::ROW>::name) << "<";
+void RowType::printChildren(std::stringstream& ss) const {
   bool any = false;
   for (size_t i = 0; i < children_.size(); ++i) {
     if (any) {
-      ss << ',';
```
Comment (mbasmanova) [inline velox/type/Type.cpp:339]: @majetideepak Deepak, this change triggered a lot of internal test failures. Can we perhaps, add another parameter "delimited" with default value of "," and set it to ", " in PlanNode::toString()?

File: velox/type/Type.cpp [hunk 4]
```diff
@@ -330,13 +330,13 @@ bool RowType::operator==(const Type& other) const {
   return true;
 }
 
-std::string RowType::toString() const {
-  std::stringstream ss;
-  ss << (TypeTraits<TypeKind::ROW>::name) << "<";
+void RowType::printChildren(
+    std::stringstream& ss,
+    const std::string_view delimiter) const {
```
Comment (mbasmanova) [inline velox/type/Type.cpp:335]: nit: drop 'const' from 'const std::string_view'; it is not needed

**Review by mbasmanova — APPROVED** [review]

**Comment by mbasmanova** [comment]
> I used delimiter instead since I remember using that before. Let me know if you prefer delimited.

Nice. I made a typo as I was thinking "delimiter".

**Review by aditi-pandit — COMMENTED** [review]
+1 Lazy review

**Pull request closed without merging**